lib/fluent/supervisor.rb

```diff
@@ -50,6 +50,9 @@ def before_run
       @rpc_server = nil
       @counter = nil
 
+      @fluentd_lockdir = Dir.mktmpdir("fluentd-lock-")
```

Comment:
How about using a block instead of manual cleanup?

```ruby
Dir.mktmpdir("fluentd-lock-") do |lockdir|
  ENV["FLUENTD_LOCKDIR"] = lockdir
  ...
end
```

Reply:
75ef92f787106d changed it so that Fluentd makes use of the automatic cleanup.
It is working on my machine.

Let's see it works nciely on other platforms as well...